client/chain/chain_test.go

```diff
@@ -51,6 +58,68 @@ func createClient(senderAddress cosmtypes.AccAddress, cosmosKeyring keyring.Keyr
 	return chainClient, err
 }
 
+func TestOfacList(t *testing.T) {
+	network := common.LoadNetwork("testnet", "lb")
+	tmClient, err := rpchttp.New(network.TmEndpoint, "/websocket")
+	assert.NoError(t, err)
+
+	senderAddress, cosmosKeyring, err := accountForTests()
+	assert.NoError(t, err)
+
+	clientCtx, err := NewClientContext(
+		network.ChainId,
+		senderAddress.String(),
+		cosmosKeyring,
+	)
+	assert.NoError(t, err)
+
+	clientCtx = clientCtx.WithNodeURI(network.TmEndpoint).WithClient(tmClient)
+
+	exchangeClient, err := exchangeclient.NewExchangeClient(network)
+	assert.NoError(t, err)
+	ctx := context.Background()
+	req := spotExchangePB.MarketsRequest{
+		MarketStatus: "active",
+	}
+	res, err := exchangeClient.GetSpotMarkets(ctx, &req)
+	assert.NoError(t, err)
+
+	marketsAssistant, err := NewMarketsAssistantInitializedFromChain(ctx, exchangeClient)
+	assert.NoError(t, err)
+
+	cc, err := createClient(senderAddress, cosmosKeyring, network)
+	assert.NoError(t, err)
+
+	defaultSubaccountID := cc.DefaultSubaccount(senderAddress)
+	marketId := res.Markets[0].MarketId
+	amount := decimal.NewFromFloat(2)
+	price := decimal.NewFromFloat(1.02)
+
+	order := cc.CreateSpotOrder(
+		defaultSubaccountID,
+		&SpotOrderData{
+			OrderType:    exchangetypes.OrderType_BUY, //BUY SELL BUY_PO SELL_PO
+			Quantity:     amount,
+			Price:        price,
+			FeeRecipient: senderAddress.String(),
+			MarketId:     marketId,
+		},
+		marketsAssistant,
+	)
+
+	msg := new(exchangetypes.MsgCreateSpotLimitOrder)
+	msg.Sender = senderAddress.String()
+	msg.Order = exchangetypes.SpotOrder(*order)
+
+	accNum, accSeq := cc.GetAccNonce()
+
+	cc.(*chainClient).ofacTxList = []string{
+		senderAddress.String(),
+	}
+	_, err = cc.BuildSignedTx(clientCtx, accNum, accSeq, 20000, msg)
+	assert.Error(t, err)
+}
```

Comment:
We should create a test that does not connect to a any node or indexer if possible. In this case we should not to create an ExchangeClient instance to create a message to test the OFAC restriction